### PR TITLE
Support importing module paths that end in trailing directory separator

### LINKED ISSUE #5533: Trailing slash on directory path breaks Import-Module command
Steps to reproduce
------------------

```powershell
PS /Users/tsulli> ipmo ./git/CUSTOMMODULE/
ipmo : The specified module './git/CUSTOMMODULE/' was not loaded because no valid module file was found in any module directory.
At line:1 char:1
+ ipmo ./git/CUSTOMMODULE/
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : ResourceUnavailable: (./git/CUSTOMMODULE/:String) [Import-Module], FileNotFoundException
+ FullyQualifiedErrorId : Modules_ModuleNotFound,Microsoft.PowerShell.Commands.ImportModuleCommand

PS /Users/tsulli> ipmo ./git/CUSTOMMODULE
PS /Users/tsulli> gmo

ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     0.2        CUSTOMMODULE
Binary     3.3.189.1  AWSPowerShell.NetCore               {Add-AASScalableTarget, Add-ACMCertificateTag, Add-ADSConfigurationItemsToApplication, Add-ASAAt...
Manifest   3.1.0.0    Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-Item, Clear-ItemProperty...}
Manifest   3.1.0.0    Microsoft.PowerShell.Utility        {Add-Member, Add-Type, Clear-Variable, Compare-Object...}
Script     1.1.7.0    PackageManagement                   {Find-Package, Find-PackageProvider, Get-Package, Get-PackageProvider...}
Script     1.5.0.0    PowerShellGet                       {Find-Command, Find-DscResource, Find-Module, Find-RoleCapability...}
Script     1.2        PSReadLine                          {Get-PSReadlineKeyHandler, Get-PSReadlineOption, Remove-PSReadlineKeyHandler, Set-PSReadlineKeyH...
```

Expected behavior
-----------------

```none
Import-Module command should succeed whether or not the trailing slash is present.
```

Actual behavior
---------------

```none
Adding a trailing slash to the directory path of a module causes Import-Module command to throw an exception.
```

Environment data
----------------

```powershell
PS /Users/tsulli> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.0.0-rc
PSEdition                      Core
GitCommitId                    v6.0.0-rc
OS                             Darwin 16.7.0 Darwin Kernel Version 16.7.0: Wed Oct  4 00:17:00 PDT 2017; root:xnu-3789.71.6~1/RELEASE_X86_64
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

Cheers,
Trevor Sullivan

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
## PR Summary

With PSReadLine as default interactive experience, it will always add a trailing directory separator character on tab-complete.  Currently, `Import-Module` fails to load the module if this character
exists as the current logic incorrectly uses `GetFileName` to get the last part of the path which returns
an empty string if the path ends in a directory separator.

Fix is to update the code path that tries to load a module given a directory path and removes the
trailing directory character, if present, so that the logic to expect the module filename to be the same
as the module directory name works.

Fix https://github.com/PowerShell/PowerShell/issues/5533

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1 [hunk 1]
```diff
@@ -30,6 +30,12 @@ Describe "Import-Module" -Tags "CI" {
         (Get-Module -Name $moduleName).Name | Should -BeExactly $moduleName
     }
 
+    It "should be able to load a module with a trailing directory separator" {
+        $modulePath = (Get-Module -ListAvailable $moduleName).ModuleBase + [System.IO.Path]::DirectorySeparatorChar
+        Import-Module -Name $modulePath
+        (Get-Module -Name $moduleName).Name | Should -BeExactly $moduleName
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1]: The module already was loaded in previous test.
Seems we could check:
```powershell
{ Import-Module -Name $modulePath -ErrorAction Stop } | Should -Not -Throw
``

Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1]: And maybe add `-Force` to really load the module and test whole code.

Comment (SteveL-MSFT) [reply test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1]: Perhaps a `Remove-Module` would make more sense here

**Review by iSazonov — APPROVED** [review]

File: src/System.Management.Automation/engine/Modules/ImportModuleCommand.cs [hunk 2]
```diff
@@ -638,6 +638,11 @@ private PSModuleInfo ImportModule_LocallyViaName(ImportModuleOptions importModul
 
                             if (!found)
                             {
+                                // If the path ends with a directory separator, remove it
```
Comment (anmenaga) [inline src/System.Management.Automation/engine/Modules/ImportModuleCommand.cs]: By looking at the code in the file, I get a feeling that it is better to place this new change (that removes trailing separator) before the call to `LoadUsingMultiVersionModuleBase` that is earlier in the file: 
`foundModule = LoadUsingMultiVersionModuleBase(rootedPath,`
This way version selection in the multi-version case will still work even with trailing separator in module path.

File: test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1 [hunk 3]
```diff
@@ -30,6 +30,13 @@ Describe "Import-Module" -Tags "CI" {
         (Get-Module -Name $moduleName).Name | Should -BeExactly $moduleName
     }
 
+    It "should be able to load a module with a trailing directory separator" {
+        $modulePath = (Get-Module -ListAvailable $moduleName).ModuleBase + [System.IO.Path]::DirectorySeparatorChar
+        Remove-Module $moduleName
```
Comment (anmenaga) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1]: Looks like module is already removed in BeforeEach.

**Review by anmenaga — APPROVED** [review]
Non-blocking comments

File: src/System.Management.Automation/engine/Modules/ImportModuleCommand.cs (hunk 2 above)
Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/Modules/ImportModuleCommand.cs]: Good catch.  I'll move it up.

**Pull request merged**